src/org/omegat/core/segmentation/SRX.java

```diff
@@ -157,32 +217,65 @@ public static SRX loadSRX(File configFile) {
                     sb.append("\n");
                 }
                 Log.logErrorRB("CORE_SRX_EXC_LOADING_SEG_RULES", sb.toString());
-                res = new SRX();
-                res.initDefaults();
-                return res;
+                return SRX.getDefault();
             }
 
             // checking the version
             if (CURRENT_VERSION.compareTo(res.getVersion()) > 0) {
                 // yeap, the segmentation config file is of the older version
 
                 // initing defaults
-                SRX defaults = new SRX();
-                defaults.initDefaults();
+                SRX defaults = SRX.getDefault();
                 // and merging them into loaded rules
                 res = merge(res, defaults);
             }
+            Log.log("using segmentation rules from " + configFile);			
         } catch (Exception e) {
             // silently ignoring FNF
             if (!(e instanceof FileNotFoundException)) {
                 Log.log(e);
             }
-            res = new SRX();
-            res.initDefaults();
+            res = SRX.getDefault();
         }
         return res;
     }
 
+    private static SRX loadSrxFile (URL rulesUrl) {
+        try {
+            List<MapRule> newMap = new ArrayList<MapRule>();
+            Srx data = (Srx) SRX_JAXB_CONTEXT.createUnmarshaller().unmarshal(rulesUrl);
+
+            for (Languagerule rules : data.getBody().getLanguagerules().getLanguagerule()) {
+
+                String lang = rules.getLanguagerulename();
+                String pattern = DEFAULT_RULES_PATTERN;
+                for (Languagemap lm : data.getBody().getMaprules().getLanguagemap()) {
+                    if (lm.getLanguagerulename().equals(rules.getLanguagerulename())) {
+                        pattern = lm.getLanguagepattern();
+                        break;
+                    }
+                }
+                List<Rule> rulesList = new ArrayList<Rule>(rules.getRule().size());
+                for (gen.core.segmentation.Rule r : rules.getRule()) {
+                    boolean isBreak = "yes".equalsIgnoreCase(r.getBreak());
+                    rulesList.add(new Rule(isBreak, r.getBeforebreak().getContent(), r.getAfterbreak()
+                            .getContent()));
+                }
+
+                newMap.add(new MapRule(lang, pattern, rulesList));
+            }
+            Log.log("using segmentation rules from " + rulesUrl);
+            // set rules only if no errors
+            SRX res = new SRX(); res.setMappingRules(newMap);
+            res.setCascade(! ((data.getHeader() != null) && ("no".equals(data.getHeader().getCascade())))); 	// in OmegaT, defaults to true
```

Comment:
Should the string comparisons be case-insensitive?

Reply:
Strictly speaking, the SRX specification does not say that values are case insensitive (if I good remember, it also contains a sample showing the contrary). I know that in the case of maprules you decided to accept them as case-insensitive, this can sound logical if we consider that ISO-639 and BCP 47 are case-insensitive, but be conscient that SRX specification does not require case-insensitivity here, meaning that you can break compatibility with SRX files coming from other tools.
Please decide.

Reply:
My instinct was that there isn't much danger in being generous in interpreting booleans, but I am satisfied by the logic put forth [here](https://stackoverflow.com/a/10511947/448068) that one ought to follow the spec, which is the XSD, and [the XSD says](https://www.gala-global.org/sites/default/files/uploads/pdfs/srx-20.xsd) `cascade` is an enumerated type with values `yes` and `no`. So the current code is fine.